### PR TITLE
Shortform on All Posts page

### PR DESCRIPTION
**Pull request by jimrandomh**
Adds shortform to the All Posts page.

Current issues:
 * When you expand a shortform comment, you still don't see its replies. Fixing this requires changing the query handling a bit, and also the way collapsing works so that there isn't a tree directly visible
 * The styling isn't very good
 * Putting the SingleLineComment hover and the PostsItem hover up against each other makes the PostsItem hover look bad

### CODE REVIEW

**Comment by jpaddison3** [comment]
I just pulled to build off of this and reduce conflicts and I noticed that when I created a shortform comment/post/item/thing and navigated to the allPosts page it didn't appear until reset.

File: packages/lesswrong/components/posts/PostsDay.jsx [hunk 1]
```diff
@@ -21,38 +20,82 @@ const styles = theme => ({
...
-        { noPosts ? (<div className={classes.noPosts}>No posts on {date.format('MMMM Do YYYY')}</div>) :
-          <div>
-            {posts.map((post, i) => <PostsItem2 key={post._id} post={post} currentUser={currentUser} index={i} />)}
-          </div>
-        }
-      </div>
-    );
+  // The most recent day is hidden if there are no posts on it, to avoid having
+  // an awkward empty partial day when it's close to midnight.
+  if (noPosts && noComments && hideIfEmpty) {
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/PostsDay.jsx]: What happens if the user is not a beta user and there are comments?

File: packages/lesswrong/components/posts/PostsDay.jsx [hunk 2]
```diff
@@ -21,38 +20,82 @@ const styles = theme => ({
...
+        <div className={classes.shortformGroup}>
+          <div className={classes.shortformTag}>
+            Shortform [Beta]
+          </div>
+          {comments?.map((comment, i) =>
+            <CommentsNode
+              comment={comment} post={comment.post}
+              key={comment._id}
+              forceSingleLine loadChildrenSeparately
+              nestingLevel={1}
```
Comment (Raemon) [inline packages/lesswrong/components/posts/PostsDay.jsx]:
```suggestion
```

File: packages/lesswrong/lib/collections/comments/views.js [hunk 3]
```diff
@@ -258,13 +258,44 @@ ensureIndex(Comments, {topLevelCommentId:1});
 // Used in findCommentByLegacyAFId
 ensureIndex(Comments, {agentFoundationsId:1});
 
+Comments.addView('topShortform', function (terms) {
+  const timeRange = ((terms.before || terms.after)
+    ? { postedAt: {
+      ...(terms.before && {$lt: new Date(terms.before)}),
+      ...(terms.after && {$gt: new Date(terms.after)})
```
Comment (Raemon) [inline packages/lesswrong/lib/collections/comments/views.js]:
```suggestion
      ...(terms.after && {$gte: new Date(terms.after)})
```

File: packages/lesswrong/components/posts/PostsDay.jsx (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/components/posts/PostsDay.jsx]: Then you get an awkward empty partial day. But that's not that bad.

**Pull request merged**